clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -119,6 +145,17 @@ public boolean ready(Node node, long now) {
     }
 
     /**
+     * Disconnect connection to a particular node (if there is one). The disconnection is asynchronous and will not be
+     * processed until the next {@link #poll(long, long) poll()} call.
+     *
+     * @param nodeId The id of the node
+     */
+    @Override
+    public void disconnect(String nodeId) {
+        selector.disconnect(nodeId);
```

Comment:
This is related to the comment above. If a client calls disconnect, it may not be interested in using this connection any more. So, we probably want to fresh the memory by removing the id from the connection state.